### PR TITLE
tikv scaling takes precedence over upgrading

### LINKED ISSUE #2739: failover and scaling are blocked if one Pod failed during rolling update
## Feature Request

**Is your feature request related to a problem? Please describe:**

Take TiKV for example, during TiKV rolling update, if one TiKV Pod failed, e.g. something wrong with the node it was running and it cannot be scheduled to any node, then the upgrade will be stuck in waiting this Pod ready and its store UP, however, in this case, if the failover cannot occur because it's blocked by the logic https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tikv_upgrader.go#L110-L115, and if users want to scale out a new TiKV to increase the replicas, it still is impossible due to the same reason.
**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 
fixes #2739 
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Support scaling and auto-failover even if a TiKV store fails in upgrading
```

### CODE REVIEW

File: tests/e2e/tidbcluster/stability.go [hunk 1]
```diff
@@ -923,13 +986,12 @@ var _ = ginkgo.Describe("[tidb-operator][Stability]", func() {
...
 			err = utiltidbcluster.WaitForTidbClusterCondition(cli, tc.Namespace, tc.Name, time.Minute*5, func(tc *v1alpha1.TidbCluster) (bool, error) {
-				exist := false
 				for _, failureStore := range tc.Status.TiKV.FailureStores {
 					if failureStore.PodName == podName {
-						exist = true
+						return true, nil
 					}
 				}
-				return exist, nil
+				return false, nil
```
Comment (cofyc) [inline tests/e2e/tidbcluster/stability.go:994]: simplify the code

File: pkg/manager/member/tikv_scaler_test.go [hunk 2]
```diff
@@ -252,19 +252,6 @@ func TestTiKVScalerScaleIn(t *testing.T) {
 			errExpectFn:   errExpectRequeue,
 			changed:       false,
 		},
-		{
-			name:          "tikv is upgrading",
```
Comment (DanielZhangQD) [inline pkg/manager/member/tikv_scaler_test.go:256]: Can we modify the case to cover the scenario that scaling can be done during upgrading?

Comment (cofyc) [reply pkg/manager/member/tikv_scaler_test.go:256]: updated

**Pull request merged**